### PR TITLE
Add DNS-SOA record.

### PR DESCRIPTION
**Pull request by Heidistein**
Bit different, this one returns a hash with the fields. This is a present from the Ruby resolver class.

### CODE REVIEW

**Comment by b4ldr** [comment]
@Heidistein I have just [merged a PR](https://github.com/voxpupuli/puppet-dnsquery/pull/114) to move all functions so that they are namespaced e.g. `dnsquery::soa` , with the old function being deprecated.  would you be able to update your function so that it uses the new namespaced form as opposed to the soon to be deprecated form.  thanks

**Comment by bastelfreak** [comment]
@Heidistein please rebase against our latest master to get rid of the merge commit.

File: spec/functions/dns_soa_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#! /usr/bin/env ruby -S rspec
```
Comment (bastelfreak) [inline spec/functions/dns_soa_spec.rb:1]: the shebang isn't required

**Comment by Heidistein** [comment]
> @Heidistein please rebase against our latest master to get rid of the merge commit.

I did a rebase, and this merge apeared from there. Did it in the webinterface of github. That was wrong, shoulf have done in the cli as usual.

**Comment by Heidistein** [comment]
fubar

**Pull request closed without merging**